tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py

```diff
@@ -0,0 +1,207 @@
+        X = self.preprocess(X, fit=True)
+        if X_val is not None:
+            X_val = self.preprocess(X_val)
+
+        if sample_weight is not None:  # TODO: support
+            logger.log(15, "sample_weight not yet supported for ImagePredictorModel, this model will ignore them in training.")
+
+        X = X.reset_index(drop=True)
+        y = y.reset_index(drop=True)
+        if X_val is not None:
+            X_val = X_val.reset_index(drop=True)
+            y_val = y_val.reset_index(drop=True)
+        X[self._label_column_name] = y
+        if X_val is not None:
+            X_val[self._label_column_name] = y_val
+
+        verbosity_image = max(0, verbosity - 1)
+        root_logger = logging.getLogger()
+        root_log_level = root_logger.level
+        # TODO: ImagePredictor doesn't use problem_type in any way at present.
+        #  It also doesn't error or warn if problem_type is not one it expects.
+        self.model = ImagePredictor(
+            problem_type=self.problem_type,
+            path=self.path,
+            # eval_metric=self.eval_metric,  # TODO: Vision only works with accuracy
+            verbosity=verbosity_image
+        )
+        classes = list(y.unique())
+        classes.sort()
+        self._classes = classes
+
+        X = ImageDataset(X, classes=classes)
+        if X_val is not None:
+            X_val = ImageDataset(X_val, classes=classes)
+
+        self.model.fit(train_data=X,
+                       tuning_data=X_val,
+                       time_limit=time_limit,
+                       hyperparameters=params,
+                       random_state=0)
+        # self.model.set_verbosity(verbosity)  # TODO: How to set verbosity of fit predictor?
+        root_logger.setLevel(root_log_level)  # Reset log level
+
+    def _predict_proba(self, X, **kwargs):
+        X = self.preprocess(X, **kwargs)
+
+        if self.problem_type == REGRESSION:
+            # FIXME: This probably won't work
+            # return self.model.predict(X, as_pandas=False)
+            raise AssertionError(f'ImagePredictorModel does not support `problem_type="{REGRESSION}"`')
+
+        y_pred_proba = self.model.predict_proba(X, as_pandas=True)
+
+        ##################
+        # FIXME: Vision should just have the standard predict_proba format by default, this is a huge amount of computation to convert
+        y_pred_proba = y_pred_proba[['score', 'id', 'image']]
+        idx_to_image_map = X[['image']]
+        idx_to_image_map = idx_to_image_map.reset_index(drop=True).reset_index(drop=False)
+        y_pred_proba = y_pred_proba.merge(idx_to_image_map, on='image')
+        y_pred_proba = y_pred_proba.drop(columns=['image'])
+        class_preds = []
+        for clss in self._classes:
+            cur_clss = y_pred_proba[y_pred_proba['id'] == clss]
+            cur_clss = cur_clss.set_index('index')['score']
+            class_preds.append(cur_clss)
+        y_pred_proba = pd.concat(class_preds, axis=1).to_numpy()
```

Comment:
Does this post-processing really need to be so complex?

@zhreshold  Should the issue be fixed in autogluon.vision/gluonCV first before merging this PR?

Reply:
Ideally, gluoncv should just return the normal pred_proba format. I don't know why it is in the format it is, but it takes a **_lot_** of work to convert it to the standard format.

Reply:
Right I guess @zhreshold can tell us what's best for now regarding this issue.

Reply:
See #1044, if the post-processing isn't a blocker then you could remove these once #1044 get merged.

Reply:
Will wait for #1044 to merge before addressing in this PR

Reply:
Code block is now removed, no longer necessary after #1044